zerver/webhooks/thinkst/view.py

```diff
@@ -0,0 +1,69 @@
+# Webhooks for external integrations.
+from typing import Any, Dict, Optional
+
+from django.http import HttpRequest, HttpResponse
+
+from zerver.decorator import api_key_only_webhook_view
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_success
+from zerver.lib.webhooks.common import check_send_webhook_message
+from zerver.models import UserProfile
+
+CANARY_MESSAGE_TEMPLATE = (
+    '**:alert: Canary{} has been triggered!**\n\n'
+    'On {}, `{}` was triggered from `{}`{}:\n\n'
+    '> {}'
+)
+
+CANARYTOKEN_MESSAGE_TEMPLATE = (
+    '**:alert: Canarytoken has been triggered on {}!**\n\n'
+    '> {} \n\n'
+    '[Manage this canarytoken]({})'
+)
+
+@api_key_only_webhook_view('Thinkst')
+@has_request_variables
+def api_thinkst_webhook(request: HttpRequest, user_profile: UserProfile,
+                        message: Dict[str, Any]=REQ(argument_type='body'),
+                        user_specified_topic: Optional[str]=REQ('topic', default=None)) -> HttpResponse:
+
+    body = None
+    topic = None
+
+    # There are two types of requests that Thinkst will send - depending on whether it is
+    # a canary or a canarytoken. Unfortunately, there isn't a great way to differentiate other
+    # than look at the contents.
+    is_canarytoken = 'Timestamp' not in message
+
+    if is_canarytoken:
+        topic = 'canarytoken alert'
+        body = CANARYTOKEN_MESSAGE_TEMPLATE.format(
+            message['time'],
+            message['memo'],
+            message['manage_url'],
+        )
+    else:
+        topic = 'canary alert - {}'.format(message['CanaryName'])
+
+        reverse_dns = ''
+        if 'ReverseDNS' in message:
+            reverse_dns = ' (`' + message['ReverseDNS'] + '`)'
+
+        name = ''
+        if user_specified_topic:
+            name = ' `' + message['CanaryName'] + '`'
```

Comment:
Would you be up for switching these to use `f-strings`?  We just migrated the whole codebase from `.format()` and `%` formatting to them, and it does read a lot nicer.

Other than this, lgtm, though maybe @Hypro999 should do a quick reivew in case I'm missing something.

Reply:
Fixed, thanks!